src/category_theory/limits/shapes/types.lean

```diff
@@ -223,4 +227,68 @@ def equalizer_limit : limits.limit_cone (parallel_pair g h) :=
 
 end fork
 
+section pullback
+open category_theory.limits.walking_pair
+open category_theory.limits.walking_cospan
+open category_theory.limits.walking_cospan.hom
+
+variables {W X Y Z : Type u}
+variables (f : X ⟶ Z) (g : Y ⟶ Z)
+
+/--
+The usual explicit pullback in the category of types, as a subtype of the product.
+The full `limit_cone` data is bundled as `pullback_limit_cone f g`.
+-/
+@[nolint has_inhabited_instance]
+abbreviation pullback_obj : Type u := { p : X × Y // f p.1 = g p.2 }
+
+-- `pullback_obj f g` comes with a coercion to the product type `X × Y`.
+example (p : pullback_obj f g) : X × Y := p
+
+/--
+The explicit pullback cone on `pullback_obj f g`.
+This is bundled with the `is_limit` data as `pullback_limit_cone f g`.
+-/
+abbreviation pullback_cone : limits.pullback_cone f g :=
+pullback_cone.mk (λ p : pullback_obj f g, p.1.1) (λ p, p.1.2) (funext (λ p, p.2))
+
+/--
+The explicit pullback in the category of types, bundled up as a `limit_cone`
+for given `f` and `g`.
+-/
+@[simps]
+def pullback_limit_cone (f : X ⟶ Z) (g : Y ⟶ Z) : limits.limit_cone (cospan f g) :=
+{ cone := pullback_cone f g,
+  is_limit := pullback_cone.is_limit_aux _
+    (λ s x, ⟨⟨s.fst x, s.snd x⟩, congr_fun s.condition x⟩)
+    (by tidy)
+    (by tidy)
+    (λ s m w, funext $ λ x, subtype.ext $
+     prod.ext (congr_fun (w walking_cospan.left) x)
+              (congr_fun (w walking_cospan.right) x)) }
+
+/--
+The pullback cone given by the instance `has_pullbacks (Type u)` is isomorphic to the
+explicit pullback cone given by `pullback_limit_cone`.
+-/
+noncomputable def pullback_cone_iso_pullback : limit.cone (cospan f g) ≅ pullback_cone f g :=
+(limit.is_limit _).unique_up_to_iso (pullback_limit_cone f g).is_limit
+
+/--
+The pullback given by the instance `has_pullbacks (Type u)` is isomorphic to the
+explicit pullback object given by `pullback_limit_obj`.
+-/
+noncomputable def pullback_iso_pullback : pullback f g ≅ pullback_obj f g :=
+(cones.forget _).map_iso $ pullback_cone_iso_pullback f g
+
+@[simp] lemma pullback_fst'
+  : (pullback_iso_pullback f g).hom ≫ (λ p, (p : X × Y).fst) = pullback.fst :=
```

Comment:
```suggestion
@[simp] lemma pullback_fst' :
  (pullback_iso_pullback f g).hom ≫ (λ p, (p : X × Y).fst) = pullback.fst :=
```

Reply:
I'm not convinced these simp lemmas will ever actually fire -- it seems unlikely the left hand side would ever appear as stated. I wonder if the elementwise version of these lemmas would be more useful.

On the other hand, the lemmas for `(pullback_iso_pullback f g).inv` should be stated, and seem fine as simp lemmas.

Reply:
Sure. I added simp lemmas for `(pullback_iso_pullback f g).inv`, and replaced these with elementwise versions.

The elementwise versions I came up with feel a bit messy to me -- but take a look and see what you think.